internal/logutil/logutil.go

```diff
@@ -0,0 +1,63 @@
+package logutil
+
+import (
+	"io/ioutil"
+	"os"
+
+	logging "github.com/ipfs/go-log"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var loggers = make(map[string]*BSLogger)
+var logAll bool
+
+type BSLogger struct {
+	*logging.ZapEventLogger
+	unsugared *zap.Logger
+}
+
+func (bsl *BSLogger) IsDebug() bool {
+	if logAll {
+		return true
+	}
+	return bsl.unsugared.Check(zap.DebugLevel, "debug log") != nil
+}
+
+func CreateLogger(name string) *BSLogger {
+	logger := logging.Logger(name)
+	bsLogger := &BSLogger{logger, logger.Desugar()}
+	loggers[name] = bsLogger
+	return bsLogger
+}
+
+func TeeLogs() *os.File {
+	tmpfile, err := ioutil.TempFile("", "tee.log")
```

Comment:
nit: I'd make this configurable.

Reply:
You mean with an environment variable?

Reply:
I was thinking `TeeLogs(filename)` but I somehow missed that this was using `ioutil.TempFile`. This is fine as-is.